### PR TITLE
ddl: wait for a while when some error occured to avoid retrying too many times. (#6601)

### PR DESCRIPTION
**Pull request by winkyao**
cherry-pick from #6601

### CODE REVIEW

**Comment by shenli** [comment]
/run-all-tests tidb-test=release-2.0 tikv=release-2.0 pd=release-2.0

**Comment by shenli** [comment]
LGTM

File: ddl/ddl_worker.go [hunk 1]
```diff
@@ -286,14 +290,23 @@ func (d *ddl) handleDDLJobQueue(shouldCleanJobs bool) error {
...
 				err = d.finishDDLJob(t, job)
 				return errors.Trace(err)
 			}
-			err = d.updateDDLJob(t, job, err != nil)
+			err = d.updateDDLJob(t, job, runJobErr != nil)
 			return errors.Trace(d.handleUpdateJobError(t, job, err))
 		})
+
+		if runJobErr != nil {
+			// wait a while to retry again. If we don't wait here, ddl will retry this job immediately,
```
Comment (zimulala) [inline ddl/ddl_worker.go]: s/ddl/DDL

File: ddl/ddl_worker.go [hunk 2]
```diff
@@ -286,14 +290,23 @@ func (d *ddl) handleDDLJobQueue(shouldCleanJobs bool) error {
...
+			// which may act like a deadlock.
+			log.Infof("[ddl] run ddl job error, sleep a while:%v then retry it.", waitTimeWhenErrorOccured)
```
Comment (zimulala) [inline ddl/ddl_worker.go]: s/ddl/DDL
Please unify the log prefix.

File: ddl/ddl_worker.go (hunk 1 above)
Comment (winkyao) [reply ddl/ddl_worker.go]: It's a cherry-pick, we should modify it in a new PR

**Pull request merged**